pkgs/os-specific/linux/input-utils/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "input-utils-${version}";
+  version = "1.3";
+  
+  src = fetchurl {
+    url = "https://www.kraxel.org/releases/input/input-${version}.tar.gz";
+    sha256 = "11w0pp20knx6qpgzmawdbk1nj2z3fzp8yd6nag6s8bcga16w6hli";
+  };
+
+  prePatch = ''
+    # Adds full path to "linux/input.h"
+    substituteInPlace ./emulate-key.c --replace "linux/input.h" "${stdenv.glibc.dev}/include/linux/input.h"
+    substituteInPlace ./input.h       --replace "linux/input.h" "${stdenv.glibc.dev}/include/linux/input.h"
+    # Use proper include path for kernel include files.
+    substituteInPlace ./name.sh       --replace "/usr/include/linux/" "${stdenv.glibc.dev}/include/linux/"
+    substituteInPlace ./lirc.sh       --replace "/usr/include/linux/" "${stdenv.glibc.dev}/include/linux/"
+  '';
+
+  makeFlags = [
+    "prefix=$(out)"
+    "STRIP="
```

Comment:
I think STRIP=strip is preferred.

Reply:
Using a bit of `ag` to grep around, I see:

 * 6 `STRIP=` (including two from packages by the same upstream author, fbida, amtterm)
 * 5 `export STRIP="${targetCC.bintools.bintools}/bin/${targetCC.bintools.targetPrefix}strip"` (ghc)
 * 2 `STRIP=-s`
 * 1 `STRIP=${stdenv.cc.bintools.targetPrefix}strip`
 * 1 `STRIP=strip`
 * 1 replacement of `STRIP=` by `#STRIP=`

And other misc:

 * -DCMAKE_STRIP=$(command -v $crossConfig-strip)

* * *

Using `STRIP=strip` leads to:

```
installing
install flags: install SHELL=/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash prefix=\$\(out\) STRIP=strip
install -d /nix/store/iw7g8gmal9bafsnsi5qxgp238chz8y3h-input-utils-1.3/bin /nix/store/iw7g8gmal9bafsnsi5qxgp238chz8y3h-input-utils-1.3/share/man/man8
install strip lsinput input-events input-kbd input-send input-recv /nix/store/iw7g8gmal9bafsnsi5qxgp238chz8y3h-input-utils-1.3/bin
install: cannot stat 'strip': No such file or directory
make: *** [GNUmakefile:39: install] Error 1
```

After looking at the source, the [mk/Variables.mk#n31](https://git.kraxel.org/cgit/input/tree/mk/Variables.mk#n31) file explains it all. That STRIP variable isn't for the strip binary, but for an option for `install`.

```
       -s, --strip
              strip symbol tables
```

The `stdenv` sets the environment STRIP to `strip`, which well, given it is used like `install $STRIP` would try to install `[./]strip` which doesn't exist.

The `STRIP=` bit was taken (IIRC) from the debian package description, or the archlinux PKGBUILD (I let that package sit in my local overlay for 5 months!).